misaligned.c

```diff
@@ -1,21 +1,56 @@
 #include <stdio.h>
+#include <string.h>
 #include <assert.h>
+#include "misaligned.h"
 
-int printColorMap() {
+//Function to print color pattern
+int printColorMap(void (*GetColorPattern)(char*, int, int)) {
     const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
     const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
     int i = 0, j = 0;
+    char colorPairNames[MAX_COLORPAIR_NAME_CHARS];
+    memset(colorPairNames, 0, MAX_COLORPAIR_NAME_CHARS);
     for(i = 0; i < 5; i++) {
         for(j = 0; j < 5; j++) {
-            printf("%d | %s | %s\n", i * 5 + j, majorColor[i], minorColor[i]);
+            sprintf(colorPairNames, "%d | %s | %s\n", i * 5 + j + 1, majorColor[i], minorColor[j]);
+            //Display color patterns
+            printf("\n%s",colorPairNames);
+            if(GetColorPattern != 0){
+                (*GetColorPattern)(colorPairNames, i, j);
+            }
         }
     }
     return i * j;
 }
 
-int main() {
-    int result = printColorMap();
-    assert(result == 25);
-    printf("All is well (maybe!)\n");
+//Get Expected color pattern
+void GetColorPattern(char* colorPattern, int majorIndex, int minorIndex){
+
+    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
+    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+    static char ExpectedColorPairNames[MAX_COLORPAIR_NAME_CHARS];
+    int numberOfMinorColors =  sizeof(minorColor) / sizeof(minorColor[0]);
+    memset(ExpectedColorPairNames, 0, MAX_COLORPAIR_NAME_CHARS);
+    sprintf(ExpectedColorPairNames, "%d | %s | %s\n", majorIndex * numberOfMinorColors + minorIndex + 1
+                                           , majorColor[majorIndex], minorColor[minorIndex]);
+    //Assertion for color pattern
+    assert(memcmp(colorPattern, ExpectedColorPairNames, MAX_COLORPAIR_NAME_CHARS) == 0);

```

Comment:
Assert happens within production code. This should not be the case. Production code should always return or output values which needs to be verified outside the scope of production code.

This assert could lead to unwanted behavior in the field.